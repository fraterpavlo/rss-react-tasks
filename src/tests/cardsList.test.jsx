import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import CardsList from 'components/cards-list';

describe('test cardsList', () => {
  let cardsDataList;
  beforeAll(() => {
    cardsDataList = [
      {
        id: 1,
        name: 'Rick Sanchez',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Male',
        origin: { name: 'Earth (C-137)', url: 'https://rickandmortyapi.com/api/location/1' },
        location: { name: 'Citadel of Ricks', url: 'https://rickandmortyapi.com/api/location/3' },
        image: 'https://rickandmortyapi.com/api/character/avatar/1.jpeg',
        url: 'https://rickandmortyapi.com/api/character/1',
        created: '2017-11-04T18:48:46.250Z',
      },
      {
        id: 2,
        name: 'Morty Smith',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Male',
        origin: { name: 'unknown', url: '' },
        location: { name: 'Citadel of Ricks', url: 'https://rickandmortyapi.com/api/location/3' },
        image: 'https://rickandmortyapi.com/api/character/avatar/2.jpeg',
        url: 'https://rickandmortyapi.com/api/character/2',
        created: '2017-11-04T18:50:21.651Z',
      },
      {
        id: 3,
        name: 'Summer Smith',
        status: 'Alive',
        species: 'Human',
        type: '',
        gender: 'Female',
        origin: {
          name: 'Earth (Replacement Dimension)',
          url: 'https://rickandmortyapi.com/api/location/20',
        },
        location: {
          name: 'Earth (Replacement Dimension)',
          url: 'https://rickandmortyapi.com/api/location/20',
        },
        image: 'https://rickandmortyapi.com/api/character/avatar/3.jpeg',
        url: 'https://rickandmortyapi.com/api/character/3',
        created: '2017-11-04T19:09:56.428Z',
      },
    ];
  });

  it('test cardsList match snapshot', () => {
    const cardsList = render(<CardsList dataList={cardsDataList} />);
    expect(cardsList).toMatchSnapshot();
  });

  it('test cardsList contain correct number of card', async () => {
    render(<CardsList dataList={cardsDataList} />);
    const cardsList = await screen.findAllByTestId('cardslist-item');
    expect(cardsList.length).toBe(cardsDataList.length);
  });

  it('test show modal detailed card after click of card', async () => {
    render(<CardsList dataList={cardsDataList} />);
    const cardOfRickSanchez = await screen.findByText('Rick Sanchez');
    fireEvent.click(cardOfRickSanchez);
    const modalDetailedCard = await screen.findByTestId('1');
    expect(modalDetailedCard).toBeInTheDocument();
  });
});
