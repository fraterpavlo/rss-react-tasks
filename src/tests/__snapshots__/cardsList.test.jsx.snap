// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test cardsList test cardsList match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="undefined cards-area__container"
      >
        <div
          class="card cards-area__item"
          data-card-id="1"
          data-testid="cardslist-item"
        >
          <span
            class="card__name"
          >
            Rick Sanchez
          </span>
          <div
            class="card__img-contain"
          >
            <img
              alt="card image"
              class="card__img"
              src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
            />
          </div>
        </div>
        <div
          class="card cards-area__item"
          data-card-id="2"
          data-testid="cardslist-item"
        >
          <span
            class="card__name"
          >
            Morty Smith
          </span>
          <div
            class="card__img-contain"
          >
            <img
              alt="card image"
              class="card__img"
              src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
            />
          </div>
        </div>
        <div
          class="card cards-area__item"
          data-card-id="3"
          data-testid="cardslist-item"
        >
          <span
            class="card__name"
          >
            Summer Smith
          </span>
          <div
            class="card__img-contain"
          >
            <img
              alt="card image"
              class="card__img"
              src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
            />
          </div>
        </div>
        <div
          class="myModal"
        >
          <div
            class="myModalContent"
          >
            <strong>
              sorry, data not found
            </strong>
            <button>
              Close
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="undefined cards-area__container"
    >
      <div
        class="card cards-area__item"
        data-card-id="1"
        data-testid="cardslist-item"
      >
        <span
          class="card__name"
        >
          Rick Sanchez
        </span>
        <div
          class="card__img-contain"
        >
          <img
            alt="card image"
            class="card__img"
            src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
          />
        </div>
      </div>
      <div
        class="card cards-area__item"
        data-card-id="2"
        data-testid="cardslist-item"
      >
        <span
          class="card__name"
        >
          Morty Smith
        </span>
        <div
          class="card__img-contain"
        >
          <img
            alt="card image"
            class="card__img"
            src="https://rickandmortyapi.com/api/character/avatar/2.jpeg"
          />
        </div>
      </div>
      <div
        class="card cards-area__item"
        data-card-id="3"
        data-testid="cardslist-item"
      >
        <span
          class="card__name"
        >
          Summer Smith
        </span>
        <div
          class="card__img-contain"
        >
          <img
            alt="card image"
            class="card__img"
            src="https://rickandmortyapi.com/api/character/avatar/3.jpeg"
          />
        </div>
      </div>
      <div
        class="myModal"
      >
        <div
          class="myModalContent"
        >
          <strong>
            sorry, data not found
          </strong>
          <button>
            Close
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
